contracts/cw20-merkle-airdrop/src/msg.rs

```diff
@@ -126,11 +127,17 @@ pub struct MigrateMsg {}
 // Signature verification is done on external airdrop claims.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct SignatureInfo {
-    pub claim_msg: SignedClaimMsg,
+    pub claim_msg: Binary,
     pub signature: Binary,
 }
-
+impl SignatureInfo {
+    pub fn extract_addr_from_memo(&self) -> Result<String, ContractError> {
```

Comment:
Best to hide the implementation details on this level
```suggestion
    pub fn extract_addr(&self) -> Result<String, ContractError> {
```